File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/workdir/src/surveycrawler/invalid_command.py
from surveycrawler.crawler_command import CrawlerCommand


class InvalidCommand(CrawlerCommand):

    # testing

    @classmethod
    def is_for(cls, a_character):
        return False

    # repeating

    def repeat(self, a_number_of_repetitions, a_crawler):
        a_crawler.signal_invalid_command()

    # executing

    def execute(self, a_crawler):
        a_crawler.signal_invalid_command()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-214358-38088/analysis/acceptance-project/src/main/java/surveycrawler/InvalidCommand.java
package surveycrawler;

public class InvalidCommand extends CrawlerCommand {

    // testing - it is the command used when no other command understands the character,
    // and also the last command when there is no command to repeat

    @Override
    public boolean isFor(char aCommand) {
        return true;
    }

    // executing

    @Override
    protected void executeOnce(SurveyCrawler aCrawler) {
        aCrawler.signalInvalidCommand();
    }
}
